src/index.test.js

```diff
@@ -0,0 +1,193 @@
+    display();
+    window.addEventListener('DOMContentLoaded', () => {
+      const checkbox = document.getElementById('checkId1');
+      checkbox.checked = true;
+      checkbox.dispatchEvent(new Event('change'));
+      const storedList = JSON.parse(localStorage.getItem('listItems12'));
+      expect(storedList[0].completed).toBe(true);
+    });
+  });
+
+  it('should update the completed status in local storage when checkbox is unchecked', () => {
+    const sampleItem = { index: 1, title: 'Sample Item', completed: true };
+    todoList.push(sampleItem);
+    display();
+    window.addEventListener('DOMContentLoaded', () => {
+      const checkbox = document.getElementById('checkId1');
+      checkbox.checked = false;
+      checkbox.dispatchEvent(new Event('change'));
+      const storedList = JSON.parse(localStorage.getItem('listItems12'));
+      expect(storedList[0].completed).toBe(false);
+    });
+  });
+
+  it('should remove all completed items from the todoList', () => {
+    const item1 = { index: 1, title: 'Item 1', completed: true };
+    const item2 = { index: 2, title: 'Item 2', completed: false };
+    const item3 = { index: 3, title: 'Item 3', completed: true };
+
+    todoList.push(item1, item2, item3);
+    display();
+    window.addEventListener('DOMContentLoaded', () => {
+      const clearBtn = document.getElementById('clsBtn');
+      clearBtn.click();
+
+      const completedItems = todoList.filter((item) => item.completed);
+      expect(completedItems.length).toBe(0);
+      expect(todoList.length).toBe(1); // Verify that incomplete items are still present in the list
+    });
+  });
+
+  // it('should remove all completed items from the todoList', () => {
+  //   const item1 = { index: 1, title: 'Item 1', completed: true };
+  //   const item2 = { index: 2, title: 'Item 2', completed: false };
+  //   const item3 = { index: 3, title: 'Item 3', completed: true };
+
+  //   todoList.push(item1, item2, item3);
+  //   display();
+  //   window.addEventListener('DOMContentLoaded', () => {
+  //   const clearBtn = document.getElementById('clsBtn');
+  //   clearBtn.click();
+  //   });
+
+  //   const completedItems = todoList.filter((item) => item.completed);
+  //   expect(completedItems.length).toBe(0);
+  // });
+
+  // it('should remove all completed items from local storage', () => {
+  //   const item1 = { index: 1, title: 'Item 1', completed: true };
+  //   const item2 = { index: 2, title: 'Item 2', completed: false };
+  //   const item3 = { index: 3, title: 'Item 3', completed: true };
+
+  //   todoList.push(item1, item2, item3);
+  //   display();
+  //   window.addEventListener('DOMContentLoaded', () => {
+  //   const clearBtn = document.getElementById('clsBtn');
+  //   clearBtn.click();
+  //   });
+  //   const storedList = JSON.parse(localStorage.getItem('listItems12'));
+  //   const completedItems = storedList.filter((item) => item.completed);
+  //   expect(completedItems.length).toBe(0);
+  // });
```

Comment:
Great work done so far Congratulations, I would just ask you to remove commented code to make your work professional otherwise everything is good